test/Tests/src/System/File_Spec.enso

```diff
@@ -359,6 +528,22 @@ spec =
                 n.read_text . should_equal "new content"
             [f, bak, n0, n1, n2, n4].each .delete
 
+        Test.specify "should create a backup when writing a dry run file with Context.Output disabled" <|
+            f = transient / "work.txt"
+            f.delete_if_exists
+            f.exists.should_be_false
+            "line 1!".write f on_problems=Report_Error . should_succeed . should_equal f
+
+            r = Context.Output.with_disabled <| "New Content!".write f
```

Comment:
Also I think just for clarity, I'd make it use the Backup mode explicitly, even though we know its the default. I guess just in case it changes in the future + the test reads easier if it's explicit.

```suggestion
            r = Context.Output.with_disabled <| "New Content!".write f on_existing_file=Existing_File_Behavior.Backup
```